lib/repl.js

```diff
@@ -326,13 +326,13 @@ function REPLServer(prompt,
     let awaitPromise = false;
     const input = code;
 
-    if (/^\s*{/.test(code) && /}\s*$/.test(code)) {
-      // It's confusing for `{ a : 1 }` to be interpreted as a block
-      // statement rather than an object literal.  So, we first try
-      // to wrap it in parentheses, so that it will be interpreted as
-      // an expression.  Note that if the above condition changes,
-      // lib/internal/repl/utils.js needs to be changed to match.
-      code = `(${code.trim()})\n`;
+    // It's confusing for `{ a : 1 }` to be interpreted as a block
+    // statement rather than an object literal.  So, we first try
+    // to wrap it in parentheses, so that it will be interpreted as
+    // an expression.  Note that if the above condition changes,
+    // lib/internal/repl/utils.js needs to be changed to match.
+    if (code.startsWith('{') && !code.endsWith(';')) {
```

Comment:
We should likely keep the regular expression. Otherwise it's not aligned anymore with the preview in case there's whitespace at the front or the end of the input.

Reply:
@BridgeAR what do you mean here? the preview code has:
```
    // For similar reasons as `defaultEval`, wrap expressions starting with a
    // curly brace with parenthesis.
    if (input.startsWith('{') && !input.endsWith(';')) {
      input = `(${input})`;
    }
```
which is what i copied over to here 🤔

Reply:
The preview input is already trimmed on both sides. This code does that afterwards.